### PR TITLE
Chapter Title Parsing

### PR DESCRIPTION
**Pull request by hokei**
I filed https://jira.nypl.org/browse/SIMPLY-3186 as a follow up 
This only deals with the most surface of the issues, there are a bunch of underlying issues that may need fixing in the future.  The good news is that it shouldn't have broken anything already working.  The bad news is that I'm punting the writing of unit tests on this in favor of finishing up features by Wednesday.

### CODE REVIEW

**Comment by kristojorg** [comment]
I'm getting two errors on the config branch when trying to run this: `cannot read href of undefined` and `cannot read length of undefined`. I tried clearing my browser data to no avail :/

**Comment by kristojorg** [comment]
Also wtf IFrameNavigator.ts is 1700 lines?! I wish I didn't see that

File: src/IFrameNavigator.ts [hunk 1]
```diff
@@ -876,16 +804,115 @@ export default class IFrameNavigator implements Navigator {
...
+              position: 0,
+            });
+          }
+        }
+      });
+
+      parentElement.appendChild(listElement);
+    };
+
+    if (manifest.toc.length) {
```
Comment (kristojorg) [inline src/IFrameNavigator.ts:877]: This might be source of the `cannot read length of undefined` error. Just a guess

**Review by kristojorg — CHANGES_REQUESTED** [review]
Reading through this code I am sorry for what you're having to do :(

**Comment by hokei** [comment]
> Also wtf IFrameNavigator.ts is 1700 lines?! I wish I didn't see that

:trash_fire_emoji:

**Review by kristojorg — APPROVED** [review]
Woohoo we are in business! It looks good to me. Some books don't have the TOC in the header, but presumably that's because they're not compatible

**Comment by hokei** [comment]
> Some books don't have the TOC in the header, but presumably that's because they're not compatible

Yeah in my initial survey I thought most of the books were epub3 but it turns out a bunch of them aren't.  I'll add them in next week.

**Pull request merged**